src/main/kotlin/com/sourcegraph/cody/initialization/EndOfTrialNotificationScheduler.kt

```diff
@@ -0,0 +1,103 @@
+package com.sourcegraph.cody.initialization
+
+import com.intellij.ide.util.PropertiesComponent
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.CurrentUserCodySubscription
+import com.sourcegraph.cody.agent.protocol.GetFeatureFlag
+import com.sourcegraph.cody.agent.protocol.Plan
+import com.sourcegraph.cody.agent.protocol.Status
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.config.ConfigUtil
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+class EndOfTrialNotificationScheduler private constructor(val project: Project) : Disposable {
+
+  private val logger = Logger.getInstance(EndOfTrialNotificationScheduler::class.java)
+
+  private val scheduler = Executors.newScheduledThreadPool(1)
+
+  init {
+    scheduler.scheduleAtFixedRate(
+        /* command = */ {
+          if (!ConfigUtil.isCodyEnabled()) {
+            return@scheduleAtFixedRate
+          }
+
+          if (project.isDisposed) {
+            this.dispose()
+          }
+
+          if (CodyAuthenticationManager.instance.getActiveAccount(project)?.isDotcomAccount() ==
+              false) {
+            return@scheduleAtFixedRate
+          }
+
+          CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+            agent.server
+                .getCurrentUserCodySubscription()
+                .thenApply { currentUserCodySubscription ->
+                  if (currentUserCodySubscription == null) {
+                    logger.warn("currentUserCodySubscription is null")
+                    return@thenApply
+                  }
+
+                  agent.server
+                      .evaluateFeatureFlag(GetFeatureFlag.CodyProTrialEnded)
+                      .completeOnTimeout(false, 4, TimeUnit.SECONDS)
```

Comment:
Just to be safe, cannot we wait longer if that is a method run once every 2h?
Or is it simply always expected to return in xx ms?

Reply:
we can for 4 seconds seems to be much more than needed for a feature flag check + it is consistent with other timeouts